### PR TITLE
The conversation functionality moved to a class

### PR DESCRIPTION
**Pull request by annando**
The conversation functionality moved into a class now. It is a draft since it should be part of the next and not the current release.

### CODE REVIEW

File: src/Module/Conversation/Community.php [hunk 1]
```diff
@@ -138,7 +138,8 @@ public static function content(array $parameters = [])
 			return $o;
 		}
 
-		$o .= conversation(DI::app(), $items, 'community', false, false, 'commented', local_user());
+//		$o .= conversation(DI::app(), $items, 'community', false, false, 'commented', local_user());
```
Comment (MrPetovan) [inline src/Module/Conversation/Community.php]:
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Conversation/Community.php (hunk 1 above)
Comment (annando) [reply src/Module/Conversation/Community.php]: Done

**Comment by annando** [comment]
There is much potential for improvements in that functionality, but I think it is a beginning.

File: src/Content/Conversation.php [hunk 2]
```diff
@@ -287,15 +287,15 @@ public function statusEditor(array $x = [], $notes_cid = 0, $popup = false)
...
-		$x['visitor']          = $x['visitor']          ?? 'block';
-		$x['is_owner']         = $x['is_owner']         ?? true;
-		$x['profile_uid']      = $x['profile_uid']      ?? local_user();
+		$x['nickname']         = $x['nickname'] ?? $user['nickname'];
+		$x['lockstate']        = $x['lockstate'] ?? ACL::getLockstateForUserId($user['uid']) ? 'lock' : 'unlock';
+		$x['acl']              = $x['acl'] ?? ACL::getFullSelectorHTML($this->page, $user['uid'], true);
+		$x['bang']             = $x['bang'] ?? '';
+		$x['visitor']          = $x['visitor'] ?? 'block';
+		$x['is_owner']         = $x['is_owner'] ?? true;
+		$x['profile_uid']      = $x['profile_uid'] ?? local_user();
```
Comment (MrPetovan) [inline src/Content/Conversation.php]: Nooo, my nice alignment!

File: src/Content/Conversation.php [hunk 3]
```diff
@@ -442,16 +442,16 @@ public function create(array $items, $mode, $update, $preview = false, $order =
...
+					. (!empty($_GET['search']) ? '&search=' . rawurlencode($_GET['search']) : '')
+					. (!empty($_GET['star']) ? '&star=' . rawurlencode($_GET['star']) : '')
+					. (!empty($_GET['order']) ? '&order=' . rawurlencode($_GET['order']) : '')
+					. (!empty($_GET['bmark']) ? '&bmark=' . rawurlencode($_GET['bmark']) : '')
+					. (!empty($_GET['liked']) ? '&liked=' . rawurlencode($_GET['liked']) : '')
+					. (!empty($_GET['conv']) ? '&conv=' . rawurlencode($_GET['conv']) : '')
+					. (!empty($_GET['nets']) ? '&nets=' . rawurlencode($_GET['nets']) : '')
+					. (!empty($_GET['cmin']) ? '&cmin=' . rawurlencode($_GET['cmin']) : '')
+					. (!empty($_GET['cmax']) ? '&cmax=' . rawurlencode($_GET['cmax']) : '')
+					. (!empty($_GET['file']) ? '&file=' . rawurlencode($_GET['file']) : '')
```
Comment (MrPetovan) [inline src/Content/Conversation.php]: What has your IDE dooooone?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Content/Conversation.php (hunk 2 above)
Comment (annando) [reply src/Content/Conversation.php]: Sorry. I wanted to keep it. But the system insisted on removing it.

File: src/Content/Conversation.php (hunk 3 above)
Comment (annando) [reply src/Content/Conversation.php]: Not my IDE, the automated check disliked it.

File: src/Content/Conversation.php (hunk 2 above)
Comment (MrPetovan) [reply src/Content/Conversation.php]: "the system" as if we couldn't change it.

Although I'm pretty sure there isn't a setting in `phpcs` to make these specific alignments.

Comment (annando) [reply src/Content/Conversation.php]: If you can make these alignments then I'm happy reverting these changes.

Comment (MrPetovan) [reply src/Content/Conversation.php]: Thank, I had a quick look at the PHPCS rules and none are related to ternary operator alignment.

**Comment by MrPetovan** [comment]
Thank you for the revert, if you keep it as is your PR will keep failing checks. It isn't a big deal for me because I know why but it may be inconvenient at some point.

**Comment by annando** [comment]
Well, I hope that you can apply the needed change to the test. Then this check won't fail anymore.

File: src/Content/Conversation.php (hunk 2 above)
Comment (annando) [reply src/Content/Conversation.php]: The check tells:
```
src/src/Content/Conversation.php (ternary_operator_spaces, binary_operator_spaces)
```
I guess that `ternary_operator_spaces` and `binary_operator_spaces` are the rules?

**Comment by MrPetovan** [comment]
I'm sorry, I looked for a suitable configuration but I couldn't find any.

**Comment by nupplaphil** [comment]
Hmmm.. There has to be a reason why it failed.. Ignoring tests is mainly the wrong way ;-) adapting it for the future is way better. Since the PR isn't part of this release, I really would prefer this, I could take a look when I'm no more I'll *gg*

**Comment by annando** [comment]
@nupplaphil my hope had been that the check is fixed until the PR is done.

**Comment by nupplaphil** [comment]
I'll try to take a look before you mark it as ready ;)

**Comment by nupplaphil** [comment]
@annando I tried to push the solution to this PR, but you had it at the 2021.09-rc branch, so it got closed automatically :X ..
If you reopen it, try this fix:
```patch
Index: .php_cs.dist
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.php_cs.dist b/.php_cs.dist
--- a/.php_cs.dist	(revision d4fd1512277b24f01da26fd0b4a57a259a595b7a)
+++ b/.php_cs.dist	(revision 7345ca02e7e5de0080cf85b1e2ed8eb656eddb2b)
@@ -33,6 +33,9 @@
 			'operators' => [
 				'=>' => 'align_single_space_minimal',
 				'='  => 'align_single_space_minimal',
+				'.'  => 'align_single_space_minimal',
+				'?'  => 'align_single_space_minimal',
+				':'  => 'align_single_space_minimal',
 			],
 		],
 		'blank_line_after_namespace'   => true,
```

**Comment by MrPetovan** [comment]
Ooooh!

**Pull request closed without merging**